### ISSUE TITLE
make amount required in AmountOfChemical

### ISSUE THREAD

**Issue by lzehl** [openMetadataInitiative/openMINDS_chemicals#11]
I think this should be required and AmountOfChemical should only be used if amount is known otherwise we should refer to category (#10 ) chemicalProduct

**Comment by Raphael-Gazzotti** [comment]
Creation of a new branch version, this change is not backward compatible.

**Comment by apdavison** [comment]
I don't think this is a good idea. AmountOfChemical is used in the property "hasPart" in ChemicalMixture, as an embedded type. The types in the chemicalProduct category are used as linked types, so we can only mix them with AmountOfChemical if we allow both linked types and embedded types in the same property. 

Semantically, "amount of chemical" is the appropriate type, even if the amount is unknown or has a large uncertainty.

One way to make this work would be to add an explicit representation for an unknown quantity, for example by having a QuantitativeValueRange with limits of zero to infinity (cf https://github.com/openMetadataInitiative/openMINDS_core/issues/330)

**Comment by lzehl** [comment]
I don't like having a schema with no properties required. This is not a good design. Let's discuss this further.

**Comment by apdavison** [comment]
It does have a required property: "chemicalProduct"

**Comment by lzehl** [comment]
@apdavison sorry I should have been more precise and my words were incorrect: What I actually meant was that a schema that is called AmountOfChemical cannot not require both the amount and the chemical. This is the whole purpose of that schema and therefore has to be required.

> Semantically, "amount of chemical" is the appropriate type, even if the amount is unknown or has a large uncertainty.

I truly disagree with this approach in our schema. A large uncertainty is still specifiable and if you don't know the amount then do not use the AmountOfChemical to specify the chemical product, then just specify the chemical product. So we may need fixing of the places where AmountOfChemical is used? @Alixbonard @lauramble could you put this together so that we can discuss concretely what to do with this issue?

**Comment by lauramble** [comment]
@lzehl Unless I'm mistaken, it seems that AmountOfChemical is only used in [ChemicalMixture](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/chemicals/chemicalMixture.schema.omi.json#L25).

ChemicalMixture, in turn, is used:
- in AmountOfChemical, property [chemicalProduct](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/chemicals/amountOfChemical.schema.omi.json#L23) (other accepted types: ChemicalSubstance, MolecularEntity)
- in Pipette, property [material](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/ephys/device/pipette.schema.omi.json#L110) (other accepted types: ChemicalSubstance, MolecularEntity)
- in PipetteUsage, properties [labelingCompound](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/ephys/device/pipetteUsage.schema.omi.json#L105) (other accepted types: ChemicalSubstance, MolecularEntity) and [pipetteSolution](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/ephys/device/pipetteUsage.schema.omi.json#L166) (only accepted type)
- in ElectrodeArray, properties [conductorMaterial](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/ephys/device/electrodeArray.schema.omi.json#L14) (other accepted types: ChemicalSubstance, MolecularEntity) and [insulatorMaterial](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/ephys/device/electrodeArray.schema.omi.json#L77) (other accepted types: ChemicalSubstance, MolecularEntity)
- in Electrode, properties [conductorMaterial](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/ephys/device/electrode.schema.omi.json#L14) (other accepted types: ChemicalSubstance, MolecularEntity) and [insulatorMaterial](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/ephys/device/electrode.schema.omi.json#L62) (other accepted types: ChemicalSubstance, MolecularEntity)
- in CellPatching, property [tissueBathSolution](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/ephys/activity/cellPatching.schema.omi.json#L252) (only accepted type)
- in TissueSampleSlicing, property [tissueBathSolution](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/specimenPrep/activity/tissueSampleSlicing.schema.omi.json#L232) (only accepted type)
- in TissueCulturePreparation, property [culturedMedium](https://github.com/openMetadataInitiative/openMINDS/blob/34ce1441979cf17dac98e43103469dea4d0b5117/schemas/latest/specimenPrep/activity/tissueCulturePreparation.schema.omi.json#L10) (only accepted type)

I think it's fair to say that AmountOfChemical is only needed in ChemicalMixture and the properties that only accept ChemicalMixture

**Comment by lzehl** [comment]
@lauramble thanks for looking this up. ChemicalMixture is embedding AmountOfChemical (which is correct) making it difficult to give the alternative option of just identifying the MolecularEntity (which would need to be linked) if the amount is unknown. 

I still believe though that both properties amount and chemicalProduct need to be required in a schema called AmountOfChemical. Based on @apdavison comment though I would assume that amount should not only allow for QV but also for QVR. We can discuss in the next meeting.

**Comment by lzehl** [comment]
Let's postpone this to after v5.0 and leave it optional right now. TBD for v6.0 or newer: split between quantitative and qualitative definitions of amount.
